### DISCUSSION TITLE
Lock node state should conform to UX standards

### DISCUSSION THREAD

**Discussion by ffxsam** [godotengine/godot-proposals#6663] [Editor]
For icon buttons, typically they work like this:

- Icon reflects the current state
- Tooltip explains what happens if clicked

The lock button satisfies the 2nd point, but not the first. I would recommend either:

- making the lock icon simply turn blue when the node is locked
- flipping the icon states (opened lock = node is not locked; closed lock = node is locked)

![CleanShot 2023-04-08 at 13 28 40](https://user-images.githubusercontent.com/12532733/230737262-22d8b81a-6ad0-421f-9aba-cbe911380251.png)

**Comment by Calinou** [comment]
> - making the lock icon simply turn blue when the node is locked
> - flipping the icon states (opened lock = node is not locked; closed lock = node is locked)

I think we can do both: flip the icon state and make the icon blue when locked.

**Reply by ffxsam** [reply]
Sounds good! I wish I could help with this, but I don't know C++.
